src/backend/distributed/commands/citus_add_local_table_to_metadata.c

```diff
@@ -333,6 +331,101 @@ CreateCitusLocalTable(Oid relationId, bool cascadeViaForeignKeys)
 }
 
 
+/*
+ * CreateCitusLocalTablePartitionOf generates and executes the necessary commands
+ * to create a table as partition of a partitioned Citus Local Table.
+ * The conversion is done by CreateCitusLocalTable.
+ */
+void
+CreateCitusLocalTablePartitionOf(CreateStmt *createStatement, Oid relationId,
+								 Oid parentRelationId)
+{
+	if (createStatement->partspec)
+	{
+		char *parentRelationName = get_rel_name(parentRelationId);
+		char *relationName = get_rel_name(relationId);
+
+		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+						errmsg("distributing multi-level partitioned tables "
+							   "is not supported"),
+						errdetail("Relation \"%s\" is partitioned table itself "
+								  "and it is also partition of relation \"%s\".",
+								  relationName, parentRelationName)));
+	}
+
+	/*
+	 * If the parent is a citus local table, we don't need distribution column.
+	 * We can do create a Citus Local Table with current table and early return.
+	 */
+	char *attachCommand = GenerateAlterTableAttachPartitionCommand(relationId);
+	List *detachCommands = list_make3(DISABLE_DDL_PROPAGATION,
+									  GenerateDetachPartitionCommand(relationId),
+									  ENABLE_DDL_PROPAGATION);
```

Comment:
Two questions:

i) Why do we need to disable/enable ddl propagation for detach command ?
ii ) Why don't we need to do this for attach command ?

Reply:
See below, I suggest to use `CascadeOperationForRelationIdList`